homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,527 @@
+    threading.Thread(target=start_web_server, daemon=True, name=DOMAIN).start()
+
+    return True
+
+
+class HueBridgeView(object):
+    """An instance of an emulated Hue bridge."""
+
+    def __init__(self, hass_instance, config):
+        """Initialize the instance."""
+        self.hass = hass_instance
+        self.cached_states = {}
+
+        self.fill_with_config(config)
+
+    def fill_with_config(self, config):
+        """Fill the instance with data from the ocnfiguration file."""
+        # Get the IP address that will be passed to the Echo during discovery
+        self.host_ip_addr = config[DOMAIN].get(CONF_HOST_IP)
+        if self.host_ip_addr is None:
+            self.host_ip_addr = util.get_local_ip()
+            _LOGGER.warning(
+                "Listen IP address not specified, auto-detected address is %s",
+                self.host_ip_addr)
+
+        # Get the port that the Hue bridge will listen on
+        self.listen_port = config[DOMAIN].get(CONF_LISTEN_PORT)
+        if not isinstance(type(self.listen_port), int):
+            self.listen_port = DEFAULT_LISTEN_PORT
+            _LOGGER.warning(
+                "Listen port not specified, defaulting to %s",
+                self.listen_port)
+
+        # Get domains that cause both "on" and "off" commands to map to "on"
+        # This is primarily useful for things like scenes or scripts, which
+        # don't really have a concept of being off
+        self.off_maps_to_on_domains = config[DOMAIN].get(
+            CONF_OFF_MAPS_TO_ON_DOMAINS)
+        if not isinstance(self.off_maps_to_on_domains, list):
+            self.off_maps_to_on_domains = DEFAULT_OFF_MAPS_TO_ON_DOMAINS
+
+        # Get whether or not entities should be exposed by default, or if only
+        # explicitly marked ones will be exposed
+        self.expose_by_default = config[DOMAIN].get(CONF_EXPOSE_BY_DEFAULT)
+        if not isinstance(type(self.expose_by_default), bool):
+            self.expose_by_default = DEFAULT_EXPOSE_BY_DEFAULT
+
+        # Get domains that are exposed by default when expose_by_default is
+        # True
+        self.exposed_domains = config[DOMAIN].get(CONF_EXPOSED_DOMAINS)
+        if not isinstance(self.exposed_domains, list):
+            self.exposed_domains = DEFAULT_EXPOSED_DOMAINS
+
+    def set_up_routes(self, app):
+        """Set up Flask routes."""
+        app.route('/description.xml', methods=['GET'])(
+            self.description_xml)
+
+        app.route('/api/<token>/lights', methods=['GET'])(
+            self.hue_api_lights)
+        app.route('/api/<token>/lights/', methods=['GET'])(
+            self.hue_api_lights)
+
+        app.route('/api/<token>/lights/<entity_id>/state', methods=['PUT'])(
+            self.hue_api_put_light)
+
+        app.route('/api/<token>/lights/<entity_id>', methods=['GET'])(
+            self.hue_api_individual_light)
+
+        app.route('/api', methods=['POST'])(

```

Comment:
The `http` component provides these routing functions already. Will Hue only work if it is run under the `/api` path?